### PR TITLE
Migrate classifications and taxa collections to API

### LINKED ISSUE #3006: migrate taxa & classifications to new API

### PR DESCRIPTION
**Pull request by cesarvarela**
closes #3006 

(The failing tests seem to be just flake, because it is not the same test that fails if look at each attemp)

### CODE REVIEW

File: site/gatsby-site/playwright/utils.ts [hunk 1]
```diff
@@ -233,4 +240,50 @@ export async function fillAutoComplete(page: Page, selector: string, sequence: s
         await page.locator(selector).pressSequentially(sequence, { delay: 500 });
         await page.getByText(target).click({ timeout: 1000 });
     }).toPass();
-}
\ No newline at end of file
+}
+
+export const mockAtlasCustomData = async (page, userId, newUserData) => {
```
Comment (cesarvarela) [inline site/gatsby-site/playwright/utils.ts:245]: @clari182 these changes will help you mock customData (and roles) client side

File: .github/workflows/test-playwright-full.yml [hunk 2]
```diff
@@ -115,13 +115,17 @@ jobs:
         working-directory: site/gatsby-site
 
       - name: Run playwright tests
-        run: npx playwright test playwright/e2e-full/ --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
+        run: npm run test:e2e:ci
```
Comment (cesarvarela) [inline .github/workflows/test-playwright-full.yml:118]: This is so we can tweak the playwright's command without having to merge the GitHub workflow.

File: site/gatsby-site/playwright.config.ts [hunk 3]
```diff
@@ -75,7 +75,7 @@ export default defineConfig({
 
   /* Run your local dev server before starting the tests */
   webServer: {
-    command: 'NODE_OPTIONS=--max-old-space-size=5600 npm run serve',
+    command: 'npx -y pm2 start npm --name "web-server" -- run serve && npx pm2 logs "web-server"',
```
Comment (cesarvarela) [inline site/gatsby-site/playwright.config.ts:78]: This line is where pm2 is used, should help with restarting the local server when it crashes

File: site/gatsby-site/server/schema.ts [hunk 4]
```diff
@@ -13,25 +13,10 @@ const gatewaySchema = stitchSchemas({
...
-                validationLevel: ValidationLevel.Off, // ditto
-            },
-            ['Incident.reports']: {
-                validationLevel: ValidationLevel.Off, // ditto
-            },
-            ['CreateVariantInput.incidentId']: {
-                validationLevel: ValidationLevel.Off // ditto
-            },
-            ['CreateVariantInput.variant']: {
+            ['Incident.editors']: {
```
Comment (kepae) [inline site/gatsby-site/server/schema.ts:16]: why array and not string as key?

File: site/gatsby-site/src/pages/apps/csettool/{mongodbAiidprodIncidents.incident_id}.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ const ToolPage = (props) => {
 
   const { data, loading } = useQuery(FIND_CLASSIFICATION, {
     variables: {
-      query: { incidents_in: { incident_id: parseInt(incident_id) }, namespace_in: allNamespaces },
+      filter: { incidents: { IN: parseInt(incident_id) }, namespace: { IN: allNamespaces } },
```
Comment (kepae) [inline site/gatsby-site/src/pages/apps/csettool/{mongodbAiidprodIncidents.incident_id}.js:19]: Any other places we have to `parseInt`? I've been looking as I go...

**Review by kepae — APPROVED** [review]
Killer. Let it rip when ready!

File: site/gatsby-site/playwright/utils.ts [hunk 6]
```diff
@@ -53,16 +54,35 @@ export const test = base.extend<TestFixtures>({
 
     login: async ({ page }, use, testInfo) => {
 
+        // TODO: this should be removed since we pass the username and password as arguments
```
Comment (pdcp1) [inline site/gatsby-site/playwright/utils.ts:57]: Please add this TODO as an issue to ensure we remember to work on this change

**Review by pdcp1 — APPROVED** [review]

File: site/gatsby-site/server/schema.ts (hunk 4 above)
Comment (cesarvarela) [reply site/gatsby-site/server/schema.ts:16]: same semantic, just different notation!

File: site/gatsby-site/src/pages/apps/csettool/{mongodbAiidprodIncidents.incident_id}.js (hunk 5 above)
Comment (cesarvarela) [reply site/gatsby-site/src/pages/apps/csettool/{mongodbAiidprodIncidents.incident_id}.js:19]: Not that I aware.........

**Pull request merged**